File: 프로그래머스/unrated/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cs
using System;

public class Solution {
    public int[] solution(int[] arr) 
    {
        int size = 1;
        while(size < arr.Length)
            size *= 2;
        
        int[] answer = new int[size];
        Array.Copy(arr, 0, answer, 0, arr.Length);
        return answer;
    }
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) {
    
    int size = arr.size();
    int pot = 1;    
    while (size > pot)
    {
        pot *= 2;
    }
    
    if (pot != size)
        arr.insert(arr.end(), pot - size, 0);
    
    return arr;
}